JASP-Engine/JASP/R/multinomialtest.R

```diff
@@ -15,546 +15,440 @@
-  }
-
-  if (perform == "run") {
-
-    state <- list()
-    state[["options"]] <- options
-    state[["chisqResults"]] <- chisqResults
-    state[["descriptivesTable"]] <- descriptivesTable
-    state[["descriptivesPlot"]] <- descriptivesPlot
-
-    return(list(results=results, status="complete", state=state,
-                keep = plotPath))
-  } else {
-    return(list(results=results, status="inited", state=state,
-                keep = plotPath))
   }
 }
 
-
-.chisquareTest <- function(dataset, options, fact, perform) {
-  # Run chi-square test and return object
+# Results function ----
+.chisquareTest <- function(jaspResults, dataset, options) {
+  # Run chi-square test and return jaspResults object
   #
   # Args:
+  #   jaspResults:
   #   dataset: input dataset
   #   options: user options
-  #   fact
-  #   perform: init or run
   #
   # Return:
-  #   Object (chisquare test)
-
+  #   No return, stores jaspResults Object (chisquare test)
+ 
+  # Take results from state if possible
+  if (!is.null(jaspResults[["stateChisqResults"]])) 
+    return(jaspResults[["stateChisqResults"]]$object)
+  
   chisqResults <- NULL
-
-  if (perform == "run" && !is.null(fact)) {
+  fact <- options$factor
+  
+  if(fact != ""){
     # first determine the hypotheses
     factorVariable <- dataset[[.v(fact)]]
+    factorVariable <- factorVariable[!is.na(factorVariable)]
     factorVariable <- as.factor(factorVariable)
     nlev <- nlevels(factorVariable)
-
+    
     if (options$counts != "") {
       counts <- dataset[[.v(options$counts)]]
       # omit count entries for which factor variable is NA
       counts <- counts[!is.na(factorVariable)]
-      # check for invalid counts
-      .checkCountsMultinomial(counts, nlev)
-      dataTable        <- counts
-      names(dataTable) <- levels(factorVariable)
-    } else {
-      dataTable <- table(factorVariable)
-    }
-
+      c <- dataset[[.v(options$counts)]]
+      factorVariable <- factor(rep(factorVariable, c), 
+                               levels = levels(factorVariable))
+    } 
+    
+    dataTable <- table(factorVariable)
+    
     hyps <- .multinomialHypotheses(dataset, options, nlev)
-
+    
     # create a named list with as values the chi-square result objects
     chisqResults <- lapply(hyps, function(h) {
       # catch warning message and append to object if necessary
-      csr <- NULL
-      warn <- NULL
-      csr <- withCallingHandlers(
-        chisq.test(x = dataTable, p = h, rescale.p = TRUE,
-                   simulate.p.value = options$simulatepval),
-        warning = function(w) {
-         warn <<- w$message
-        }
+      csr  <- warn <- NULL
+      # need to improve this try statement
+      csr  <- withCallingHandlers(
+                chisq.test(x = dataTable, p = h, rescale.p = TRUE,
+                           simulate.p.value = options$simulatepval),
+                warning = function(w) warn <<- w$message
       )
       csr[["warn"]] <- warn
```

Comment:
You're not using these warning messages. Previously they were added as footnotes under the table.